docs/Superoperator representations.md

```diff
@@ -12,49 +16,565 @@ where $i$ is a row and $j$ is a column index.
+
+
+When representing quantum channels one insight is used may times.
+
+Consider two Hilbert spaces $\mathbb H_A$ and $\mathbb H_B$ with dimensions $d_A$ and $d_B$ respectively. An abstract quantum process matrix $\mathcal Q$ lives in the combined (bipartite) space of $\mathbb H_A \otimes \mathbb H_B$ so $\mathcal Q$ is a $d_A^2\times d_B^2$ matrix.
+
+We can represent the process as a tensor with components
+
+$$\mathcal Q_{m,\mu;n,\nu} = \langle m, \mu |\mathcal Q |n,\nu \rangle $$ 
+where $|n,\nu\rangle = |n\rangle \otimes |\nu\rangle$, $m,n\in \{0,\ldots, d_A-1\}$, $\mu,\nu\in \{0,\ldots, d_B-1\}$ and all vectors are in the standard basis.
+
+With respect to these indicies some useful operations are [GRAPTN]:
+ 
+Transpose $T$: $\mathcal Q_{m,\mu;n,\nu} \mapsto \mathcal Q_{n,\nu;m,\mu},$  
+SWAP: $\mathcal Q_{m,\mu;n,\nu} \mapsto \mathcal Q_{\mu,m;\nu,n},$  
+Row-reshuffling $R_r$: $\mathcal Q_{m,\mu;n,\nu} \mapsto \mathcal Q_{m,n;\mu,\nu},$  
+Col-reshuffling $R$: $\mathcal Q_{m,\mu;n,\nu} \mapsto \mathcal Q_{\nu,\mu;n,m}.$
+
+The importance of understanding reshuffling can be understood as understanding the relationship between 
+$${\rm vec}(G)\otimes {\rm vec}(\Gamma) \quad {\rm and} \quad  {\rm vec}(G\otimes\Gamma)$$
+where $G$ and $\Gamma$ are matrices that act on $\mathbb H_A$ and $\mathbb H_B$ respectively, as explaind in [VECQO].
+
+### A note on numerical implementations
+Most linear algebra (or tensor) libraries have the ablity to `reshape` a matrix and `swapaxes` (or sometimes it is called `permute_dims`). 
+
+If you are trying to reshuffle indicies usually the first job is to write your matrix in tensor form. This requires reshaping a $d_A^2\times d_B^2$ matrix into a $d_A\times d_A\times d_B \times d_B$ tensor. Next you `permute1_dims` or `swapaxes`. Often $d_A = d_B$ so we `reshape` to a Matrix that has the same dimensions as the orignal  $d_A^2\times d_A^2$ matix.
+
+
+## The $n$-qubit Pauli basis 
+ The $n$-qubit Pauli basis is denoted $\mathcal P^{\otimes n} $ where $\mathcal  P = \{ I, X, Y, Z \}$ are the usual Pauli matricies. It is an operator baisis for the $d = 2^n$ dimensional Hilbert space and there are $d^2 = 4^n$ operators in $\mathcal P^{\otimes n} $. If one divides all the operators by $\sqrt{d}$ the basis is orthonormal with respect to the Hilbert-Schmidt inner product.
+ 
+It is often convenient to index the $d^2$ operators with a single label, e.g. $P_1=I^{\otimes n},\, \ldots,\, P_{d^2}= Z^{\otimes n}$  (or $P_0=I^{\otimes n}$ if you like zero indexing). In anycase, as these operators are Hermitian and unitary they obey $P_i^2=I^{\otimes n}$.
+ 
+To be explicit, for two qubits $d=4$ and we have 16 operators e.g. $\{II, IX, IY, IZ, XI, XX, XY, ..., ZZ\}$ were $II$ should be intepreted as $I\otimes I$ etc. The single index would be $\{P_1, P_2, P_3, P_4, P_5, P_6, P_7, ..., P_{16}\}$.
+
+## Quantum channels in the Kraus decomposition (or operator-sum representation)
```

Comment:
It would be worthwhile to point out that Kraus representation is not unique.  This is important when people want to compare representations.  Sets of operators that look very different can actually represent the same operator (in the same basis even).

Reply:
It is also worth mentioning a canonical choice: Kraus operators that are proportional to the unvectorized eigenvectors of the Choi Matrix (proportionality being set by the square root of the corresponding eigenvalue).

Reply:
Point 1. This is a subtle point. Many people think about the non-uniqueness of Kraus operators in terms of the lack of the unitary degree of freedom in the POVM elements. If your Kraus operators are derived from physics they are unique.  That can be seen in a post measurement state, not simply the statistics of outcomes. If they arise from converting a different representation they many not be unique. Anyway good point, I'll add a brief comment.

Point 2. Again it depends where you got your Kraus operators from.